### PR TITLE
fix build failure with boost-1.58.0 by using higher revision

### PR DESCRIPTION
**Pull request by srekolam**

### CODE REVIEW

**Comment by adamjstewart** [comment]
Is there a reason we can't use a range of versions?

**Comment by haampie** [comment]
It'll likely work with the latest boost

**Comment by mwkrentel** [comment]
@srekolam Yes, please change this to a range.  I know other packages
(dyninst, hpctoolkit) that have a problem with specifically 1.68.0

**Comment by srekolam** [comment]
@mwkrentel  , @haampie , the latest version of boost-1.75.0 does not work with the package.. i tried using depends_on('boost@', type='link'), it is not working.. the boost-1.58.0 version , has an issue with one of the patches which is why i tried using specific versions..
i see depends_on('boost@:1.68.0', type='link') works.. l can make the changes..

**Comment by alalazo** [comment]
In case there are specific versions of a dependency that are problematic, we should be able to express that like:
```python
conflicts('^boost@1.68.0', when='...', msg='...')
```
If that doesn't work, I would consider it a bug and fix it for the ASP based solver.

**Comment by srekolam** [comment]
@haampie ,can you please review.
@adamjstewart , can you please reveiw

File: var/spack/repos/builtin/packages/miopen-hip/package.py [hunk 1]
```diff
@@ -25,12 +25,14 @@ class MiopenHip(CMakePackage):
...
     depends_on('cmake@3:', type='build')
-    depends_on('boost@1.58.0', type='link')
+    depends_on('boost@1.67.0:1.73.0', type='link')
     depends_on('pkgconfig', type='build')
     depends_on('bzip2', type='link')
     depends_on('sqlite', type='link')
     depends_on('half', type='build')
     depends_on('zlib', type='link', when='@3.9.0:')
+    conflicts('^boost@1.74.0')
+    conflicts('^boost@1.75.0')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/miopen-hip/package.py]: These two conflicts aren't needed if the dependency if limited to a range finishing at 1.73.0

**Review by adamjstewart — APPROVED** [review]

**Comment by srekolam** [comment]
@haampie  , can you please review and let me know if its good ?

**Pull request merged**